Doc/whatsnew/3.12.rst

```diff
@@ -856,6 +856,19 @@ although there is currently no date scheduled for their removal.
   :keyword:`for`, :keyword:`if`, :keyword:`in`, :keyword:`is` and :keyword:`or`.
   In a future release it will be changed to a syntax error. (:gh:`87999`)
 
+* The following :mod:`ast` features have been deprecated in documentation since
+  Python 3.8, now cause a :exc:`DeprecationWarning` to be emitted at runtime
+  when they are accessed or used, and may be removed in a future version of Python:
+
+  * :class:`!ast.Num`
+  * :class:`!ast.Str`
+  * :class:`!ast.Bytes`
+  * :class:`!ast.NameConstant`
+  * :class:`!ast.Ellipsis`
+
+  Use :class:`ast.Constant` instead.
+  (Contributed by Serhiy Storchaka in :gh:`90953`.)
```

Comment:
If you like:

```suggestion
  (Contributed by Serhiy Storchaka and Alex Waygood in :gh:`90953`.)
```

Reply:
I don't feel like I deserve this much credit; Serhiy did the vast majority of the work here :)